src/lib/Lib/MrDocsCompilationDatabase.cpp

```diff
@@ -165,6 +167,40 @@ adjustCommandLine(
     return new_cmdline;
 }
 
+std::optional<std::string> 
+executeCmakeExportCompileCommands(llvm::StringRef cmakePath, llvm::StringRef cmakeListsPath) 
+{
+    if ( ! llvm::sys::fs::exists(cmakePath)) {
```

Comment:
Return expected<std::string, Error> and describe the problem instead of `std::nullopt`?